sklearn/impute/_iterative.py

```diff
@@ -234,6 +236,26 @@ class IterativeImputer(_BaseImputer):
            [10.       ,  4.9999...,  9.        ]])
     """
 
+    _parameter_constraints = {
+        **_BaseImputer._parameter_constraints,
+        "estimator": [None, HasMethods(["predict"])],
+        "sample_posterior": ["boolean"],
+        "max_iter": [Interval(Integral, -1, None, closed="neither")],
+        "tol": [Interval(Real, 0, None, closed="both")],
+        "n_nearest_features": [None, Interval(Integral, 0, None, closed="neither")],
+        "initial_strategy": [
+            StrOptions({"mean", "median", "most_frequent", "constant"})
+        ],
+        "imputation_order": [
+            StrOptions({"ascending", "descending", "roman", "arabic", "random"})
+        ],
+        "skip_complete": ["boolean"],
+        "min_value": [None, Real, "array-like"],
+        "max_value": [None, Real, "array-like"],
+        "verbose": ["verbose"],
+        "random_state": ["random_state"],
+    }
+
```

Comment:
I see the following code in `_get_ordered_idx`:

```python
else:
            raise ValueError(
                "Got an invalid imputation order: '{0}'. It must "
                "be one of the following: 'roman', 'arabic', "
                "'ascending', 'descending', or "
                "'random'.".format(self.imputation_order)
            )
```

I think that we can remove it now.